engine/gamesys/src/dmsdk/gamesys/script.h

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 The Defold Foundation
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DMSDK_GAMESYS_SCRIPT_H
+#define DMSDK_GAMESYS_SCRIPT_H
+
+#include <dmsdk/dlib/buffer.h>
+#include <dmsdk/script/script.h>
+#include <dmsdk/gameobject/gameobject.h>
+
+extern "C"
+{
+#include <dmsdk/lua/lua.h>
+#include <dmsdk/lua/lauxlib.h>
+}
+
+
+namespace dmScript
+{
+    /*#
+     * Get current game object instance
+     * Works in both gameobjects and gui scripts
+     *
+     * @param L [type: lua_State*] lua state
+     * @return instance [type: dmGameObject::HInstance]
+     */
+    dmGameObject::HInstance CheckGOInstance(lua_State* L);
+
+    /*#
+     * Get gameobject instance
+     *
+     * The instance reference (url) at stack index "index" will be resolved to an instance.
+     * @note The function only accepts instances in "this" collection. Otherwise a lua-error will be raised.
+     *
+     * @param L [type: lua_State*] lua state
+     * @param index [type: int] lua-arg
+     * @return instance [type: lua_State*] gameobject instance
+     *
+     * @examples
+     *
+     * How to get the position of a gameobject in a script extension
+     *
+     * ```cpp
+     * static int get_position(lua_State* L)
+     * {
+     *     DM_LUA_STACK_CHECK(L, 3);
+     *     dmGameObject::HInstance instance = dmScript::CheckGOInstance(L, 1);
+     *     dmVMath::Point3 position = dmGameObject::GetPosition(instance);
+     *     lua_pushnumber(L, position.getX());
+     *     lua_pushnumber(L, position.getY());
+     *     lua_pushnumber(L, position.getZ());
+     *     return 3;
+     * }
+     * ```
+     */
+    dmGameObject::HInstance CheckGOInstance(lua_State* L, int index);
+
+    /*# buffer ownership
```

Comment:
The following code was previously in the _lower_ level library include file `script/script.h` which is imho not a good solution (include file in one lib, source code in another)